### PR TITLE
chore(cli): simplify `setNested()` and `hasNested()`

### PR DESCRIPTION
**Pull request by timreichen**

### CODE REVIEW

**Review by iuioiua — APPROVED** [review]
Thanks

File: cli/parse_args.ts [hunk 1]
```diff
@@ -373,12 +373,11 @@ function setNested(
   value: unknown,
   collect = false,
 ) {
-  keys.slice(0, -1).forEach((key) => {
-    object[key] ??= {};
-    object = object[key] as NestedMapping;
-  });
+  keys = [...keys];
+  const key = keys.pop();
+  if (!key) throw new Error(`'keys' cannot be an empty array.`);
```
Comment (kt3k) [inline cli/parse_args.ts:378]: This check seems always false

Comment (iuioiua) [reply cli/parse_args.ts:378]: Ah, yes. Are you suggesting we just use a non-null assertion?

Comment (kt3k) [reply cli/parse_args.ts:378]: ah, wait, it might be fine. Let me check

Comment (kt3k) [reply cli/parse_args.ts:378]: > Ah, yes. Are you suggesting we just use a non-null assertion?

Sounds good. That was what the previous code was doing at `const key = keys.at(-1)!`

**Pull request merged**